### PR TITLE
Di 22468

### PR DESCRIPTION
**Pull request by leonPep42**

### CODE REVIEW

File: addon.config.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
       "papi_data_index": "1.0.11",
       "data_visualization": "0.7.3",
       "slugs": "1.0.22",
-      "slideshow": "1.0.5"
+      "slideshow": "1.0.5",
```
Comment (ShirMarciano) [inline addon.config.json:27]: slideshow is not a valid dependency
I didn't notice you added it
why do you need it?

File: client-side/src/app/settings/configuration-assistant/configuration-assistant.component.ts [hunk 2]
```diff
@@ -98,6 +99,8 @@ export class ConfigurationAssistantComponent implements OnInit {
           for(const name in transaction_lines_scheme.Fields) {
             if(transaction_lines_scheme.Fields[name].Type == 'Integer' || transaction_lines_scheme.Fields[name].Type == 'Double')
               this.transactionLinesFieldsOptions.push({ Key: name, Value: name });
+            if(name.includes('Item.'))
+              this.itemCategoryOptions.push({ Key: name, Value: name });
```
Comment (ShirMarciano) [inline client-side/src/app/settings/configuration-assistant/configuration-assistant.component.ts:103]: Starts with?

File: server-side/installation.ts [hunk 3]
```diff
@@ -41,13 +43,102 @@ export async function install(client: Client, request: Request): Promise<any> {
 }
 
 export async function uninstall(client: Client, request: Request): Promise<any> {
+    try {
+        const service = new MyService(client);
+        await service.papiClient.post(`/user_defined_collection/schemes/UserTarget/hard_delete`);
+        await service.papiClient.post(`/user_defined_collection/schemes/AccountTarget/hard_delete`);
```
Comment (ShirMarciano) [inline server-side/installation.ts:49]: please put it in a function called - deleteUDCs

File: server-side/installation.ts [hunk 4]
```diff
@@ -41,13 +43,102 @@ export async function install(client: Client, request: Request): Promise<any> {
...
+        throw new Error(`Failed to delete UDCs. error - ${err}`);
+    }
+    
     return {success:true,resultObject:{}}
 }
 
 export async function upgrade(client: Client, request: Request): Promise<any> {
+    try {
+        const service = new MyService(client);
+        await createAbstractSchemes(service);
```
Comment (ShirMarciano) [inline server-side/installation.ts:60]: this code should be run only in upgrade to 0.6 version

File: server-side/installation.ts [hunk 5]
```diff
@@ -41,13 +43,102 @@ export async function install(client: Client, request: Request): Promise<any> {
...
+        await createUDCs(service);
+    } catch (err) {
+        throw new Error(`Failed to create Schemes. error - ${err}`);
+    }
```
Comment (ShirMarciano) [inline server-side/installation.ts:64]: this code needs to run in upgrade from version under 0.6

File: server-side/installation.ts [hunk 6]
```diff
@@ -41,13 +43,102 @@ export async function install(client: Client, request: Request): Promise<any> {
...
 }
 
 export async function downgrade(client: Client, request: Request): Promise<any> {
     return {success:true,resultObject:{}}
 }
+
+
+async function createAbstractSchemes(service) {
+    await service.papiClient.post(`/addons/data/schemes/${AddonUUID}`,{
+        Name: "userTarget",
```
Comment (ShirMarciano) [inline server-side/installation.ts:75]: the name of abstract schema needs to be in snake case (according to Ido) - user_target

**Review by ShirMarciano — COMMENTED** [review]
@leonPep42 please review my comments

File: addon.config.json (hunk 1 above)
Comment (leonPep42) [reply addon.config.json:27]: Eyal asked for it because his pages files contains slideshow block

Comment (ShirMarciano) [reply addon.config.json:27]: ok, we need to notify @ido-pepperi that it need to be added to the dependency addons list

File: server-side/installation.ts [hunk 7]
```diff
@@ -119,26 +117,36 @@ async function createAbstractSchemes(service) {
...
 }
 
-async function createUDCs(service) {
-    const allUdcsNames = (await service.papiClient.get(`/user_defined_collection/schemes`)).map(udc => udc.Name);
+async function createUDCs(service: MyService) {
+    const allUdcsNames = (await service.papiClient.get(`/user_defined_collections/schemes`)).map(udc => udc.Name);
     if(!allUdcsNames.includes("UserTarget")) {
-        await service.papiClient.post(`/user_defined_collection/schemes`, {
+        await service.papiClient.post(`/user_defined_collections/schemes`, {
             Name: "UserTarget",
```
Comment (ShirMarciano) [inline server-side/installation.ts:124]: I see in the api design on the name:
"Name": "my_table", // camelCase (or snake_case) Not PascalCase

Please verify with @roiaharon-pepperi  that you UDC name is valid (even if he didn't block you in his API)

File: server-side/installation.ts [hunk 8]
```diff
@@ -119,26 +117,36 @@ async function createAbstractSchemes(service) {
...
+                Name: "user_target"
             },
             Description: "Target for user"
         });
     }
     if(!allUdcsNames.includes("AccountTarget")) {
-        await service.papiClient.post(`/user_defined_collection/schemes`, {
+        await service.papiClient.post(`/user_defined_collections/schemes`, {
             Name: "AccountTarget",
             Extends: {
```
Comment (ShirMarciano) [inline server-side/installation.ts:135]: same comment as above about the UDC name

**Review by ShirMarciano — COMMENTED** [review]
@leonPep42  please see new comments

File: server-side/installation.ts (hunk 7 above)
Comment (leonPep42) [reply server-side/installation.ts:124]: where did you see it? Roi said only PascalCase is valid as UDC name

Comment (ShirMarciano) [reply server-side/installation.ts:124]: In the api design of the schema
I will approve it and will talk with Ido about how the api design is not enough

**Pull request merged**